# Calculate format=diff pkg(sys-kernel/geek-sources)>=3.13&&merge(x11-drivers/nvidia-drivers)>=334
diff -Nuar nvidia-drivers-334/kernel/nv-acpi.c nvidia-drivers-334-new/kernel/nv-acpi.c
--- nvidia-drivers-334/kernel/nv-acpi.c	2014-02-18 21:55:10.335718157 +0400
+++ nvidia-drivers-334-new/kernel/nv-acpi.c	2014-02-18 21:51:25.000000000 +0400
@@ -15,6 +15,10 @@
 #include "nv-linux.h"
 #include "nv-reg.h"
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0) && defined(ACPI_HANDLE)
+#define DEVICE_ACPI_HANDLE(a) ACPI_HANDLE(a)
+#endif
+
 #if defined(NV_LINUX_ACPI_EVENTS_SUPPORTED)
 static RM_STATUS   nv_acpi_extract_integer (const union acpi_object *, void *, NvU32, NvU32 *);
 static RM_STATUS   nv_acpi_extract_buffer  (const union acpi_object *, void *, NvU32, NvU32 *);
@@ -303,7 +307,10 @@
 
     if (pNvAcpiObject->notify_handler_installed)
     {
+		#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0)
+ 		/* beginning with 3.13, acpi_remove_notify_handler() waits for events to finish */
         NV_ACPI_OS_WAIT_EVENTS_COMPLETE();
+		#endif
 
         // remove event notifier
         status = acpi_remove_notify_handler(device->handle, ACPI_DEVICE_NOTIFY, nv_acpi_event);
